### PR TITLE
Struct attribute handle

### PR DESCRIPTION
**Pull request by jeizsm**
Handle struct attribute. So it is not needed to add attribute to every field in struct, what can be cumbersome in big structs.
And multiple refactors to remove duplication mostly

### CODE REVIEW

File: src/generate.rs [hunk 1]
```diff
@@ -14,11 +15,54 @@ pub enum GenMode {
...
+            ..
+        })) => {
+            if ident == meta_name {
+                let visibility = Ident::new(&s.value(), Span::call_site());
+                Some(visibility)
+            } else {
+                None
+            }
+        }
+        // `#[get(vis = "pub")]` or `#[set(vis = "pub")]`
```
Comment (Hoverbear) [inline src/generate.rs]: This seems unrelated, maybe you meant this bit for another PR?

File: src/lib.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@ These macros are not intended to be used on fields which require custom logic in
 extern crate getset;
 
 #[derive(Getters, Setters, MutGetters, Default)]
+#[get = "pub"] #[set = "pub"] #[get_mut = "pub"]
```
Comment (Hoverbear) [inline src/lib.rs]: Can you instead add another example that explicitly describes how the struct-level option interacts with field level options. I think this can highlight the feature and caveats better.

File: src/lib.rs [hunk 3]
```diff
@@ -31,6 +31,23 @@ fn main() {
     assert_eq!(*foo.private(), 2);
 }
 ```
+```compile_fail
```
Comment (Hoverbear) [inline src/lib.rs]: This needs some related documentation or something. :)

**Comment by Hoverbear** [comment]
Hi @jeizsm !

Thanks **so much** for doing some cleaning! :)

I think this is a reasonable feature request and am in favor of merging it.

I'd ask you to improve the documentation around the feature this adds, and allow the `get(vis = "pub")` feature to happen in a different PR.

@boscop, PTAL

File: src/generate.rs (hunk 1 above)
Comment (jeizsm) [reply src/generate.rs]: Yeah, I just forgot to remove it

File: src/lib.rs (hunk 2 above)
Comment (jeizsm) [reply src/lib.rs]: I think compile_fail example perfect for that

File: src/lib.rs (hunk 3 above)
Comment (jeizsm) [reply src/lib.rs]: Added documentation about struct level atrributes here

Comment (Hoverbear) [reply src/lib.rs]: I think the `compile_fail` could be reduced to this:

```rust
    #[derive(Getters, Default)]
    #[get = "pub"]
    pub struct Foo {
        #[get]
        private: i32,
        public: i32,
    }
```

Then maybe we can have a full, valid example of the struct level attributes and field level working?

Comment (jeizsm) [reply src/lib.rs]: It's compile_fail test not because we cannot mix struct and field level attributes. It's fine.
But generated getter is private, and we try to call it and this is compile time error.
I clarify this is with comment
>Sorry to be such a stickler about this
It's fine

**Comment by Hoverbear** [comment]
Closing in favor of #24. :)

**Comment by Hoverbear** [comment]
This feature is now merged, thanks for the contribution!
